### ISSUE TITLE
Disconnect - Respawn - When dying and respawning, the player often gets a disconnect screen

### ISSUE THREAD

**Issue by NickPSF** [psforever/PSF-LoginServer#365]
When dying and respawning, the player often gets a disconnect screen.

**Comment by Mazo** [comment]
```
2020-04-17 18:15:09,649  INFO "sessionId=69" WorldSessionActor - SpawnRequestMessage: SpawnRequestMessage(0,AMS,0,0,3)
2020-04-17 18:15:09,649  INFO "" WorldSessionActor - Zone.Lattice.SpawnPoint: spawn point on z3 at ams 3135 selected
2020-04-17 18:15:09,649  INFO "" WorldSessionActor - Load in zone z3 at position Vector3(640.0474,2426.9763,54.40347) in 5 seconds
2020-04-17 18:15:09,763  INFO "sessionId=62" WorldSessionActor - SpawnRequestMessage: SpawnRequestMessage(0,Facility,0,0,3)
2020-04-17 18:15:09,763  INFO "" WorldSessionActor - Zone.Lattice.SpawnPoint: spawn point on z3 in building 21 selected
2020-04-17 18:15:09,763  INFO "" WorldSessionActor - Load in zone z3 at position Vector3(602.6424,2408.79,45.56358) in 10 seconds
2020-04-17 18:15:10,990  INFO "sessionId=65" WorldSessionActor - ObjectHeld: ObjectHeldMessage(ValidPlanetSideGUID(4664),255,false)
2020-04-17 18:15:12,508  INFO "sessionId=65" WorldSessionActor - Chat: ChatMsg(CMT_VOICE,true,,EFE,None)
2020-04-17 18:15:13,006  INFO "sessionId=65" WorldSessionActor - AvatarImplantMessage: AvatarImplantMessage(ValidPlanetSideGUID(0),Activation,1,1)
2020-04-17 18:15:13,656  WARN "sessionId=71" WorldSessionActor - UseItem: anticipated a Kit ValidPlanetSideGUID(6065), but can't find it
2020-04-17 18:15:14,152  WARN "sessionId=71" WorldSessionActor - UseItem: anticipated a Kit ValidPlanetSideGUID(6065), but can't find it
2020-04-17 18:15:14,669  INFO "" WorldSessionActor - Player VS Mazo z3-9274 0/100 0/50 is registered
2020-04-17 18:15:14,670  INFO "" WorldSessionActor - Player Mazo will respawn
2020-04-17 18:15:14,670  INFO "" WorldSessionActor - Setting up combat engineering UI ...
2020-04-17 18:15:14,673  INFO "" services.teamwork.SquadService - Actor[akka://PsLogin/user/world-udp-endpoint/world-session-router/world-session-69#-1259106959] has left /VS/Squad
2020-04-17 18:15:14,673  INFO "" world-session-router - Dropping session ID=69 (reason: world-session-69 died)
2020-04-17 18:15:14,675 ERROR "sourceThread=PsLogin-akka.actor.default-dispatcher-22, akkaSource=akka://PsLogin/user/world-udp-endpoint/world-session-router/world-session-69, sourceActorSystem=PsLogin, akkaTimestamp=18:15:14.673UTC" akka.actor.OneForOneStrategy - null
java.lang.NullPointerException: null
	at WorldSessionActor$$anonfun$HandleSetCurrentAvatar$1.apply$mcVI$sp(WorldSessionActor.scala:3356)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:160)
	at WorldSessionActor.HandleSetCurrentAvatar(WorldSessionActor.scala:3350)
	at WorldSessionActor$$anonfun$Started$1.applyOrElse(WorldSessionActor.scala:1213)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:484)
	at WorldSessionActor.akka$actor$MDCContextAware$$super$aroundReceive(WorldSessionActor.scala:78)
	at akka.actor.MDCContextAware$class.aroundReceive(MDCContextAware.scala:25)
	at WorldSessionActor.aroundReceive(WorldSessionActor.scala:78)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

This is down to the line `player.Actor ! Player.ImplantInitializationStart(slot)`

Somehow we need to ensure the player's actor is initialized before we can send messages to it, or hold the messages in a mailbox to be delivered when ready

**Comment by NickPSF** [comment]
Still occurring on 51000, 4/18/20

**Comment by Mazo** [comment]
Fixed by #373?

**Comment by NickPSF** [comment]
I think this is fixed, closing. https://github.com/psforever/PSF-LoginServer/issues/382 is another respawn issue which is different, I believe. Will re-open this if needed.
